Neos.ContentRepository.Core/Classes/Feature/WorkspacePublication/Event/WorkspaceWasPublished.php

```diff
@@ -58,6 +59,11 @@ public static function fromArray(array $values): self
         );
     }
 
+    public function getWorkspaceName(): WorkspaceName
+    {
+        return $this->sourceWorkspaceName;
+    }
+
```

Comment:
Ohhh,... hmm. This seems to be dangerous. In your case "workspaceName" is "sourceWorkspaceName" but in other cases its "targetWorkspaceName". That's why we didn't implelemt the interface here earlier.

Also, is that needed at any point?